### PR TITLE
[MNT-21638] Script task execution

### PR DESCRIPTION
**Pull request by tiagosalvado10**

### CODE REVIEW

File: repository/src/main/java/org/alfresco/service/cmr/repository/ScriptProcessor.java [hunk 1]
```diff
@@ -73,7 +73,17 @@ public interface ScriptProcessor extends Processor
...
      */
     public Object executeString(String script, Map<String, Object> model);
-    
+
+    /**
+     * Execute script string
+     *
+     * @param script    the script string
+     * @param model     the context model
+     * @param secure    the flag that indicates if string is considered secure to be executed
```
Comment (jottley) [inline repository/src/main/java/org/alfresco/service/cmr/repository/ScriptProcessor.java:82]: We use the variable name `secure` in multiple locations.  Can we make sure that what it means is adequately described in the javadoc? Someone coming to this new may not understand what we mean by "is considered secure to be executed"

File: repository/src/main/java/org/alfresco/service/cmr/workflow/WorkflowService.java [hunk 2]
```diff
@@ -77,7 +77,24 @@
...
-    
+
+    /**
+     * Deploy a Workflow Definition to the Alfresco Repository
+     *
+     * @param  engineId  the bpm engine id
+     * @param  workflowDefinition  the workflow definition
+     * @param  mimetype  the mimetype of the workflow definition
+     * @param  name  a name representing the deployment
+     * @parm   fullAccess true if workflow should be considered secure and have full access during its execution
```
Comment (jottley) [inline repository/src/main/java/org/alfresco/service/cmr/workflow/WorkflowService.java:88]: full access to what?

File: repository/src/main/java/org/alfresco/repo/workflow/activiti/ActivitiWorkflowEngine.java [hunk 3]
```diff
@@ -242,45 +242,45 @@ public List<WorkflowInstance> cancelWorkflows(List<String> workflowIds)
...
+        String localId = createLocalId(workflowId);
+        try
+        {
+            ProcessInstance processInstance = runtimeService.createProcessInstanceQuery().processInstanceId(localId).singleResult();
+            if (processInstance == null)
+            {
+                throw new WorkflowException(messageService.getMessage(ERR_CANCEL_UNEXISTING_WORKFLOW));
+            }
+
+            // TODO: Cancel VS delete?
```
Comment (balsarori) [inline repository/src/main/java/org/alfresco/repo/workflow/activiti/ActivitiWorkflowEngine.java:256]: Deleting a running process instance cancels the process execution. So, here delete and cancel are the same thing.

Comment (tiagosalvado10) [reply repository/src/main/java/org/alfresco/repo/workflow/activiti/ActivitiWorkflowEngine.java:256]: The method cancelWorkflow(String) and the TODO comment were already there before this PR. Git has assumed as new code but it isn't.

File: packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/workflow/deployments/GetDeploymentsSanityTests.java [hunk 4]
```diff
@@ -35,7 +35,7 @@ public void getNonNetworkDeploymentsWithAdmin() throws JsonToModelConversionExce
         restClient.assertStatusCodeIs(HttpStatus.OK);
         deployments.assertThat().entriesListIsNotEmpty();
         deployments.getOneRandomEntry().onModel().assertThat()
-                .fieldsCount().is(3).and()
+                .fieldsCount().is(4).and()
```
Comment (tpage-alfresco) [inline packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/workflow/deployments/GetDeploymentsSanityTests.java:38]: I would just delete checks like this. They're not useful.

Comment (tpage-alfresco) [reply packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/workflow/deployments/GetDeploymentsSanityTests.java:38]: Oh - unless the idea here is to say "There are only the fields id, deployedAt and name".  What is the 4th field and can we just check the value of it?

**Review by jamalkm — COMMENTED** [review]
Good work. I left a minor JavaDoc issues that need fixing!

File: repository/src/main/java/org/alfresco/service/cmr/repository/ScriptProcessor.java (hunk 1 above)
Comment (tiagosalvado10) [reply repository/src/main/java/org/alfresco/service/cmr/repository/ScriptProcessor.java]: Description has been improved.

File: repository/src/main/java/org/alfresco/service/cmr/workflow/WorkflowService.java (hunk 2 above)
Comment (tiagosalvado10) [reply repository/src/main/java/org/alfresco/service/cmr/workflow/WorkflowService.java]: Description has been improved.

File: repository/src/main/java/org/alfresco/repo/processor/ScriptServiceImpl.java [hunk 5]
```diff
@@ -277,10 +296,12 @@ protected Object execute(ScriptProcessor processor, String location, Map<String,
...
      * @param script    the script string
-     * @param model     the context model
+     * @param model     the context model
+     * @param secure    the flag indicating if string script is considered secure (e.g., if it comes from classpath)
+     *                  if true it will have access to the full execution context, if false the script will be executed in a sandbox context
      * @return Object   the result of the script 
      */
-    protected Object executeString(ScriptProcessor processor, String script, Map<String, Object> model)
+    protected Object executeString(ScriptProcessor processor, String script, Map<String, Object> model, boolean secure)
     {
```
[the remark this answers is not in the document]
Comment (tiagosalvado10) [reply repository/src/main/java/org/alfresco/repo/processor/ScriptServiceImpl.java:309]: The `processor` parameter has been documented in all missing places.

File: repository/src/main/java/org/alfresco/repo/workflow/WorkflowComponent.java [hunk 6]
```diff
@@ -74,6 +74,17 @@
      * @since 4.0
      */
     public WorkflowDeployment deployDefinition(InputStream workflowDefinition, String mimetype, String name);
+
+    /**
+     * Deploy a Workflow Definition
+     *
+     * @param workflowDefinition  the content object containing the definition
+     * @param mimetype (optional)  the mime type of the workflow definition
+     * @param name (optional)  a name to represent the deployment
```
[the remark this answers is not in the document]
Comment (tiagosalvado10) [reply repository/src/main/java/org/alfresco/repo/workflow/WorkflowComponent.java:83]: `fullAcess` has been added.

File: repository/src/main/java/org/alfresco/repo/workflow/WorkflowComponent.java [hunk 7]
```diff
@@ -86,7 +97,15 @@
...
      */
     public boolean isDefinitionDeployed(InputStream workflowDefinition, String mimetype);
-    
+
+    /**
+     * Sets the deployment category if applicable to allow the workflow to have full access
+     *
+     * @param workflowDefinition  the definition to check
+     * @param fullAccess          true if category should be defined in order to consider the deployment secure
+     */
```
[the remark this answers is not in the document]
Comment (tiagosalvado10) [reply repository/src/main/java/org/alfresco/repo/workflow/WorkflowComponent.java:106]: `fullAcess` parameter has been removed.

**Review by jamalkm — APPROVED** [review]

File: packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/workflow/deployments/GetDeploymentsSanityTests.java (hunk 4 above)
Comment (tiagosalvado10) [reply packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/workflow/deployments/GetDeploymentsSanityTests.java:38]: Added `category` field to the unit test.

**Pull request merged**